core/commands/config.go

```diff
@@ -362,29 +398,44 @@ func buildProfileHelp() string {
 	return out
 }
 
-func transformConfig(configRoot string, configName string, transformer config.Transformer) error {
+func transformConfig(configRoot string, configName string, transformer config.Transformer, dryRun bool) (*config.Config, *config.Config, error) {
 	r, err := fsrepo.Open(configRoot)
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
 	defer r.Close()
 
 	cfg, err := r.Config()
 	if err != nil {
-		return err
+		return nil, nil, err
+	}
+
+	oldCfg := *cfg
+
+	if dryRun {
+		newCfg := *cfg
+		err = transformer(&newCfg)
+		if err != nil {
+			return nil, nil, err
+		}
+		return &oldCfg, &newCfg, nil
 	}
 
 	err = transformer(cfg)
```

Comment:
We're repeating the `transformer` call (with the same `*cfg` argument I believe). It would seem to me that the actual difference of using the `dry-run` option is whether or not we're persisting the new config (`BackupConfig`/`SetConfig` calls), maybe that can be moved to the `if` and avoid the early `return`. I think that would clarify the `cfg`/`oldCfg`/`newCfg` uses, we just stick to `oldCfg`/`newCfg` nomenclature. Does that make sense?

Reply:
I think `oldCfg := *cfg` is required. This makes a copy of global `config.Config`.If we don't make a copy, we would update the global `config.Config` unintentionally.
Because `fsrepo.Open()`  returns a global `repo.Repo` instance and `Repo.Config()` returns a point that points to a global `config.Config`.

Reply:
> We're repeating the `transformer` call (with the same `*cfg` argument I believe). It would seem to me that the actual difference of using the `dry-run` option is whether or not we're persisting the new config (`BackupConfig`/`SetConfig` calls), maybe that can be moved to the `if` and avoid the early `return`. I think that would clarify the `cfg`/`oldCfg`/`newCfg` uses, we just stick to `oldCfg`/`newCfg` nomenclature. Does that make sense?

I think `oldCfg := *cfg` is required. This makes a copy of global `config.Config`.If we don't make a copy, we would update the global `config.Config` unintentionally.
Because `fsrepo.Open()`  returns a global `repo.Repo` instance and `Repo.Config()` returns a point that points to a global `config.Config`.

Reply:
It's my pleasure.

Reply:
> Good point! Could you add a couple of comments to the function clarifying that please?

It's my pleasure.